src/Zypper.cc

```diff
@@ -488,6 +491,58 @@ namespace {
     return false;
   }
 
+  bool checkRequiredCapabilities ( Zypper & zypper, GlobalOptions &gopts_r )
+  {
+    switch ( zypper.command().toEnum() ) {
+      case ZypperCommand::UPDATE_e:
+      case ZypperCommand::INSTALL_e:
+      case ZypperCommand::SRC_INSTALL_e:
+      case ZypperCommand::PATCH_e:
+      case ZypperCommand::DIST_UPGRADE_e:
+      case ZypperCommand::INSTALL_NEW_RECOMMENDS_e:
+      case ZypperCommand::VERIFY_e:
+      case ZypperCommand::REMOVE_e:
+      case ZypperCommand::ADD_REPO_e:
+      case ZypperCommand::REMOVE_REPO_e:
+      case ZypperCommand::MODIFY_REPO_e:
+      case ZypperCommand::RENAME_REPO_e:
+      case ZypperCommand::ADD_SERVICE_e:
+      case ZypperCommand::REMOVE_SERVICE_e:
+      case ZypperCommand::MODIFY_SERVICE_e: {
+
+        struct statfs fsinfo;
+        memset( &fsinfo, 0, sizeof(struct statfs) );
+
+        int err = 0;
+        do {
+          err = statfs( gopts_r.root_dir.c_str(), &fsinfo );
+        } while ( err == -1 && errno == EINTR );
+
+        if ( !err ) {
+          if ( fsinfo.f_flags & ST_RDONLY ) {
+
+            bool isTransactionalServer = ( fsinfo.f_type == BTRFS_SUPER_MAGIC && PathInfo( "/usr/sbin/transactional-update" ).isFile() );
```

Comment:
good idea, but we have no test suite around for that, so it would be best if you could provide that via PR, just make sure to add a cmake switch to enable it for builds where it is relevant